darts/dataprocessing/transformers/static_covariates_transformer.py

```diff
@@ -0,0 +1,366 @@
+            if isinstance(tr_out_cat, csr_matrix):
+                tr_out_cat = tr_out_cat.toarray()
+
+        transformed_df = StaticCovariatesTransformer._reshape_out(
+            series,
+            (tr_out_num, tr_out_cat),
+            component_mask=(component_mask_num, component_mask_cat),
+            col_map_cat=col_map_cat,
+        )
+
+        return series.with_static_covariates(transformed_df)
+
+    @staticmethod
+    def ts_inverse_transform(series: TimeSeries, *args, **kwargs) -> TimeSeries:
+        transformer_num, transformer_cat = args
+        component_mask_num, component_mask_cat = kwargs.get("component_mask")
+        col_map_cat = kwargs.get("col_map_cat")
+
+        vals_num, vals_cat = StaticCovariatesTransformer._reshape_in(
+            series, component_mask=(component_mask_num, component_mask_cat)
+        )
+        tr_out_num, tr_out_cat = None, None
+        if sum(component_mask_num):
+            tr_out_num = transformer_num.inverse_transform(vals_num)
+        if sum(component_mask_cat):
+            tr_out_cat = transformer_cat.inverse_transform(vals_cat)
+
+        transformed_df = StaticCovariatesTransformer._reshape_out(
+            series,
+            (tr_out_num, tr_out_cat),
+            component_mask=(component_mask_num, component_mask_cat),
+            col_map_cat=col_map_cat,
+        )
+
+        return series.with_static_covariates(transformed_df)
+
+    def _transform_iterator(
+        self, series: Sequence[TimeSeries]
+    ) -> Iterator[Tuple[TimeSeries, Any, Any]]:
+        # use numerical and categorical transformers for 'ts_transform()'
+        return zip(
+            series,
+            [self.scaler_num] * len(series),
+            [self.scaler_cat] * len(series),
+        )
+
+    def _inverse_transform_iterator(
+        self, series: Sequence[TimeSeries]
+    ) -> Iterator[Tuple[TimeSeries, Any, Any]]:
+        # use numerical and categorical transformers for 'ts_inverse_transform()'
+        return zip(
+            series,
+            [self.scaler_num] * len(series),
+            [self.scaler_cat] * len(series),
+        )
+
+    @staticmethod
+    def _reshape_in(
+        series: TimeSeries,
+        component_mask: Optional[Tuple[np.ndarray, np.ndarray]] = None,
+    ) -> Tuple[np.array, np.array]:
+        # we expect component mask to be (numeric component mask, categorical component mask)
+        assert component_mask is not None
+        component_mask_num, component_mask_cat = component_mask
+
+        # returns tuple of (numeric static covariates, categorical static covariates)
+        vals = series.static_covariates_values(copy=False)
+        return vals[:, component_mask_num], vals[:, component_mask_cat]
+
+    @staticmethod
+    def _reshape_out(
+        series: TimeSeries,
+        vals: Tuple[np.ndarray, np.ndarray],
+        component_mask: Optional[Tuple[np.ndarray, np.ndarray]] = None,
+        col_map_cat: Optional[Dict[str, str]] = None,
```

Comment:
I did it like this because otherwise it raised a warning that the signature doesn't match the base classes signature.
Both options are not really great :D

Reply:
OK then, we can leave with it :)